libp2p/nameresolving/dnsresolver.nim

```diff
@@ -140,9 +147,14 @@ method resolveTxt*(
   for _ in 0 ..< self.nameservers.len:
     let server = self.nameservers[0]
     try:
-      let response = await getDnsResponse(server, address, TXT)
-      trace "Got TXT response", server = $server, answer=response.answers.mapIt(it.toString())
-      return response.answers.mapIt(it.toString())
+      let
+        response = await getDnsResponse(server, address, TXT)
+        answers = response.answers.mapIt(
+            try: it.toString()
+            except Exception as exc: raise newException(CatchableError, exc.msg)
```

Comment:
same, needs to be `Defect`

Reply:
the way to write these is:
```
try: ...
except CatcahbleError as exc: raise exc
except Exception as exc: raiseAssert exc.msg
```
unless you know that dns raises something that inherits from Exception, in which case it gets more complicated:
```
try: ...
except CatcahbleError as exc: raise exc
except Defect as exc: raise exc
except Exception as exc: raise (ref CatchableError)(msg: exc.msg, parent: exc)
```